### DISCUSSION TITLE
10x10 UBL HS -- Compiling Firmware Manually

### DISCUSSION THREAD

**Discussion by HermanGaviriaMarin** [Jyers/Marlin#1331] [Q&A] [answered]
### **Task at hand**: Compile Ender 3v2 firmware to support 10x10 UBL BL-Touch High Speed.  Also enable linear advance (which I believe I can do on my own, I need help with the high speed 10x10 UBL grid).

_**Current Experience:**_ I have compiled my own firmware before, I am not an expert but I have been able to compile firmware with custom Max temperatures (above 260C) and also choose different thermistor cartridge.

_**What I know:**_ I know that I must either uncomment some lines or introduce new ones, or changed existing ones. I have been searching through the source code for some line that resembles what I need, but I have not found one that specifically looks like a switch in between high speed probing and normal slow probing, (well I did see some lines but I am not sure what to do exactly because both lines were not comments).

**_How to help me:_** LOL what do I need to change? My goal is to use the template for UBL BL-Touch 10x10 and turn it into a 10x10 High speed, and if its possible how do I turn the 10x10 grid into a 15x15 UBL High speed.

![image](https://user-images.githubusercontent.com/84948613/130335887-f81f4662-d484-44a0-866b-2aa15cca1426.png)
![image](https://user-images.githubusercontent.com/84948613/130335908-6e51e289-cc4e-4834-b81b-1ce63e28664a.png)

I know that I could just sub to the Patreon and get the service, but I wanna see if I can do it with the help of the community.

Thx for your time and help.

**Comment by hitboxr** [comment, marked as answer]
Grid size for UBL is defined by `#define GRID_MAX_POINTS_X` in `Configuration.h`. Setting this to 15 will convert to a 15x15 mesh.
High Speed probing mode is controlled by `#define BLTOUCH_HS_MODE` in `Configuration_adv.h`. Uncommenting this line will enable stow-less probing.

**Reply by HermanGaviriaMarin** [reply]
So "Stow-less Probing" is the technical name Marling uses for what we call High Speed probing? Also Thank you so much for referencing the specific lines and things I need to change, very much appreciated

**Reply by hitboxr** [reply]
No I just called it that because that's the only difference between normal probing and "high speed". Instead of stowing the probe between each measurement, it simply leaves the probe extended the entire time and just takes measurements from it. It's less precise (sometimes drastically) but it is faster. To be honest unless you're rebuilding your mesh suuuper frequently I'm not convinced it's worth it for the decrease in accuracy, but it depends on your use case for sure.
